src/timedate/timedated.c

```diff
@@ -890,6 +890,11 @@ static int method_set_time(sd_bus_message *m, void *userdata, sd_bus_error *erro
         } else
                 timespec_store(&ts, (usec_t) utc);
 
+        /* when utc time is before systemd build date time*/
+        /* convert epoch time to microseconds to compare with utc*/
+        if (utc <= ((int64_t)TIME_EPOCH * USEC_PER_SEC))
```

Comment:
as mentioned please don't use `utc` at this point anymore, but look at `ts.tv_sec` instead. The `timespec_store()` lines above have already converted the 64bit usec_t into a timespec where the seconds part is accessible in `ts.tv_sec`. Hence you can shorten this line to:

```c
if (ts.tv_sec < TIME_EPOCH)
        …
```